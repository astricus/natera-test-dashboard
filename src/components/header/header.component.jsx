import React from 'react';
import { connect } from 'react-redux';
import { selectMode } from '../../redux/mode/mode.selectors';
import { setMode } from '../../redux/mode/mode.actions';
import Container from '../grid/container/container.component';

import './header.styles.scss';

const Header = ({ mode, setMode }) => (
  <div className="header">
    <Container>
      <ul className="header__navigation navigation">
        <li>
          <button
            className={`navigation__item ${
              mode ? 'navigation__item--active' : ''
            }`}
            onClick={() => setMode(true)}
          >
            Dashboard mode
          </button>
        </li>
        <li>
          <button
            className={`navigation__item ${
              !mode ? 'navigation__item--active' : ''
            } d-non-xs`}
            onClick={() => setMode(false)}
          >
            Edit mode
          </button>
        </li>
      </ul>
    </Container>
  </div>
);

const mapStateToProps = (state) => ({
  mode: selectMode(state),
});

const mapDispatchToProps = (dispatch) => ({
  setMode: (mode) => dispatch(setMode(mode)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
