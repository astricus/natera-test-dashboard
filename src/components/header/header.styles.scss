// Media-queries mixins
@import '../../styles/mixins.scss';

.header {
  height: 202px;
  background-color: var(--color-primary);

  &__navigation {
    padding-top: 47px;
  }

  @include respond(phone) {
    height: 75px;

    &__navigation {
      padding-top: 0;
    }
  }
}

.navigation {
  margin-left: 0;
  margin-bottom: 0;
  width: 100%;
  display: flex;
  justify-content: space-between;
  list-style: none;

  &__item {
    position: relative;
    display: block;
    font-family: 'Montserrat';
    text-decoration: none;
    font-size: 20px;
    font-weight: bold;
    line-height: 24px;
    background-color: transparent;
    border: none;
    cursor: pointer;
    color: rgba(255, 255, 255, 0.5);
    &--active {
      color: rgba(255, 255, 255, 1);
    }
    &--active::after {
      content: '';
      position: absolute;
      left: 0;
      bottom: -11px;
      width: 100%;
      border-bottom: 3px solid #ffffff;
    }
    &:focus {
      outline: none;
    }

    @include respond(phone) {
      margin-top: 24px;
      margin-left: 24px;
      cursor: default;

      &--active::after {
        border-bottom: none;
      }
    }
  }
}
