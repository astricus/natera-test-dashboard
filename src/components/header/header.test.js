import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './header.component';

describe('Header component', () => {
  const mockSetMode = jest.fn();
  const mode = true;
  const mockProps = {
    mode: mode,
    setMode: mockSetMode,
  };

  const headerWrapper = shallow(<Header {...mockProps} />);

  test('should render Header component', () => {
    expect(headerWrapper).toMatchSnapshot();
  });

  test('should call setMode when Edit Mode button is clicked', () => {
    headerWrapper.find('button').at(1).simulate('click');
    expect(mockSetMode).toHaveBeenCalled();
  });
});
