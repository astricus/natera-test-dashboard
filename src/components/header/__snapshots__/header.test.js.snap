// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header component 1`] = `
<div
  className="header"
>
  <Container>
    <ul
      className="header__navigation navigation"
    >
      <li>
        <button
          className="navigation__item navigation__item--active"
          onClick={[Function]}
        >
          Dashboard mode
        </button>
      </li>
      <li>
        <button
          className="navigation__item  d-non-xs"
          onClick={[Function]}
        >
          Edit mode
        </button>
      </li>
    </ul>
  </Container>
</div>
`;
